### PR TITLE
Add delivery option and start date utils

### PR DESCRIPTION
**Pull request by remybach**
🐿 v2.12.4

## Feature Description

Adds delivery option and delivery start date utils as well as fixing up the custom validation stuff to work with async/await and allow markup in error messages.

## Link to Ticket / Card:

https://trello.com/c/m1k5q0pA/1320-8-delivery-page-delivery-start-date

## Has the necessary documentation been created / updated?
- [x] Yes

## Has this been given a review by Design/UX?
- [x] Not required for this ticket

### CODE REVIEW

File: tests/utils/delivery-start-date.spec.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		let startDateUtil;
+		let startDateChangeResult;
+
+		async function setup () {
+			fetchMock.mock('/foo/bar', {
+				firstDeliveryDate: '2019-04-13',
+				firstDeliveryDateString: 'Saturday 13th of April 2019'
+			});
+			startDateUtil = new DeliveryStartDate(document);
+			startDateChangeResult = await startDateUtil.handleDeliveryStartDateChange('/foo/bar', () => {
```
Comment (ludrob) [inline tests/utils/delivery-start-date.spec.js]:
```suggestion
			startDateChangeResult = await startDateUtil.handleDeliveryStartDateChange('/api/path', () => {
```
Shall we have something more descriptive, I find using foo/bar everywhere makes it hard to read. 
In this case, we both have foo/bar in the url and the returned payload just below.

File: utils/delivery-option.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Utility for the `n-conversion-forms/partial/delivery-option.html` partial
+ * @example
+ * const deliveryOption = new DeliveryOption(document);
+ */
+class DeliveryOption {
+	/**
+	 * Initalise the Email utility
```
Comment (ludrob) [inline utils/delivery-option.js]: Email utility? copy paste leftover?
```suggestion
	 * Initalise the DeliveryOption utility
```

File: utils/delivery-start-date.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+const fetchres = require('fetchres');
+
+/**
+ * Utility for the `n-conversion-forms/partial/delivery-start-date.html` partial
+ * @example
+ * const deliveryStartDate = new DeliveryStartDate(document);
+ */
+class DeliveryStartDate {
+	/**
+	 * Initalise the Email utility
```
Comment (ludrob) [inline utils/delivery-start-date.js]:
```suggestion
	 * Initalise the DeliveryStartDate utility
```

File: utils/delivery-option.js (hunk 2 above)
Comment (remybach) [reply utils/delivery-option.js]: 🤦‍♂ Oops... good catch

**Pull request merged**